AAT_android/app/src/main/java/com/ase/aat_android/activities/CoursesActivity.java

```diff
@@ -0,0 +1,169 @@
+package com.ase.aat_android.activities;
+
+import android.app.Activity;
+import android.app.ListActivity;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Color;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.BaseAdapter;
+import android.widget.TextView;
+
+import com.ase.aat_android.R;
+import com.ase.aat_android.data.Course;
+import com.ase.aat_android.util.Constants;
+import com.ase.aat_android.util.EndpointsURL;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.android.gms.appindexing.Action;
+import com.google.android.gms.appindexing.AppIndex;
+import com.google.android.gms.appindexing.Thing;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.tasks.RuntimeExecutionException;
+
+import org.restlet.data.MediaType;
+import org.restlet.data.Method;
+import org.restlet.resource.ClientResource;
+import org.restlet.resource.Resource;
+import org.restlet.resource.ResourceException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CoursesActivity extends ListActivity {
+    private class RetrieveCoursesTask extends BaseAsyncTask<Boolean, Boolean, ArrayList<Course>> {
+
+        public RetrieveCoursesTask(Activity activity) {
+            super(activity);
+        }
+
+        @Override
+        protected ArrayList<Course> doInBackground(Boolean... params) {
+            String url = EndpointsURL.HTTP_ADDRESS+ EndpointsURL.REQUEST_COURSES;
+            ClientResource coursesRetRes = new ClientResource(Method.GET, url);
+            coursesRetRes.setResponseEntityBuffering(true);
+            coursesRetRes.setRequestEntityBuffering(true);
+            coursesRetRes.accept(MediaType.APPLICATION_ALL_JSON);
+
+            ObjectMapper mapper = new ObjectMapper();
+            List<Object> courseObjects = null;
+            try {
+                courseObjects = mapper.readValue(coursesRetRes.get().getText(), new TypeReference<List<com.aat.datastore.Course>>() {
```

Comment:
I tried before something like: 
List<Group> courseObjects =  mapper.readValue(coursesRetRes.get().getText(), new TypeReference<List<Object>>() 

In that case you wouldn't need the method ArrayList<Course> retrieveCourses(List<Object> courseObjects).